src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,68 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.Commands;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.command.ListCommand;
+import duke.command.DeleteCommand;
+import duke.command.ByeCommand;
+import duke.command.FindCommand;
+import duke.exception.CommandInvalidException;
+
+public class Parser {
+
+    /**
+     * Returns validity whether the command given.
+     * If the command is valid, return true.
+     * If the command is invalid, return false.
+     *
+     * @param no given command.
+     * @return validity of the command.
+     *
+     */
+    public static boolean isValid(String no) { //check if the given command is valid //throw something
+        String[] inputs = no.split(" ");
+        for (Commands com : Commands.values()) {
+            if (com.name().equalsIgnoreCase(inputs[0])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns an instatiate of Command for the given command.
+     *
+     * @param command given command.
+     * @return an instatiate of Command.
+     * @throws CommandInvalidException If command is invalid.
+     */
+    public static Command parse(String command) throws CommandInvalidException {
+        if (isValid(command)) {
+            String[] inputs = command.split(" ");
```

Comment:
Are these comments necessary? I believe it is clear what these if else statements in your code are intending to do.

Reply:
Okay, will delete it soon.

Reply:
Okay, will delete it soon.